go/vt/vttablet/tabletserver/planbuilder/plan_test.go

```diff
@@ -72,7 +72,76 @@ func TestPlan(t *testing.T) {
 			var err error
 			statement, err := sqlparser.Parse(tcase.input)
 			if err == nil {
-				plan, err = Build(statement, testSchema)
+				plan, err = Build(statement, testSchema, false)
+			}
+			PassthroughDMLs = false
+
+			var out string
+			if err != nil {
+				out = err.Error()
+			} else {
+				bout, err := json.Marshal(plan)
+				if err != nil {
+					t.Fatalf("Error marshalling %v: %v", plan, err)
+				}
+				out = string(bout)
+			}
+			if out != tcase.output {
+				t.Errorf("Line:%v\ngot  = %s\nwant = %s", tcase.lineno, out, tcase.output)
+				if err != nil {
+					out = fmt.Sprintf("\"%s\"", out)
+				} else {
+					bout, _ := json.MarshalIndent(plan, "", "  ")
+					out = string(bout)
+				}
+				fmt.Printf("\"%s\"\n%s\n\n", tcase.input, out)
+			}
+		})
+	}
+}
+
+func TestPlanPoolUnsafe(t *testing.T) {
+	testSchema := loadSchema("schema_test.json")
+	for tcase := range iterateExecFile("pool_unsafe_cases.txt") {
+		t.Run(tcase.input, func(t *testing.T) {
+			var plan *Plan
+			var err error
+			statement, err := sqlparser.Parse(tcase.input)
+			require.NoError(t, err)
+			// In Pooled Connection, plan building will fail.
+			plan, err = Build(statement, testSchema, false /* isReservedConn */)
+			require.Error(t, err)
+			out := err.Error()
+			if out != tcase.output {
+				t.Errorf("Line:%v\ngot  = %s\nwant = %s", tcase.lineno, out, tcase.output)
+				if err != nil {
+					out = fmt.Sprintf("\"%s\"", out)
+				} else {
+					bout, _ := json.MarshalIndent(plan, "", "  ")
+					out = string(bout)
+				}
+				fmt.Printf("\"%s\"\n%s\n\n", tcase.input, out)
+			}
+			// In Reserved Connection, plan will be built.
+			plan, err = Build(statement, testSchema, true /* isReservedConn */)
+			require.NoError(t, err)
+			require.NotEmpty(t, plan)
```

Comment:
I would really like to see the plan in the txt file, not just checking that it's not empty.

Reply:
It is vttablet plan. I can add it. I believe VTGate side plan is of more interest.